src/hotspot/share/asm/codeBuffer.cpp

```diff
@@ -97,12 +97,12 @@ CodeBuffer::CodeBuffer(CodeBlob* blob) DEBUG_ONLY(: Scrubber(this, sizeof(*this)
 
 void CodeBuffer::initialize(csize_t code_size, csize_t locs_size) {
   // Compute maximal alignment.
-  int align = _insts.alignment();
+  int align = _consts.alignment() + _insts.alignment() + _stubs.alignment();
```

Comment:
I suggest to clean this up by introducing a variable:
```
int total_size = code_size + _consts.alignment() + _insts.alignment() + _stubs.alignment() + 3 * slop;
```
You explicitly list all three sections. So you have disconnected `SEC_LIMIT`. Or you can add an assert checking that `SEC_LIMIT` is 3.